src/UiPath.Workflow/Activities/ExpressionCompiler.cs

```diff
@@ -35,6 +35,19 @@ protected static Assembly GetAssemblyForType(ITypeSymbol type)
         protected static Type GetSystemType(ITypeSymbol typeSymbol, Assembly assembly)
         {
             StringBuilder typeName = new();
+
+            if (typeSymbol is INamedTypeSymbol namedTypeSymbol)
+            {
+                if (namedTypeSymbol.IsGenericType)
+                {
+                    var arguments = namedTypeSymbol.TypeArguments.Select(t => GetSystemType(t, GetAssemblyForType(t))).ToArray();
```

Comment:
Also test if this is a generic of a type in coded (e.g.) `IEnumerable<CodedType>`

Reply:
Tested it now, it works.